src/threading.h

```diff
@@ -879,7 +862,8 @@ template<class mpz_type> bool gcd_unsigned(
     data.threshold=(uint64*)&threshold[0];
 
     data.uv_counter_start=c_thread_state.counter_start+counter_start_delta+1;
-    data.out_uv_counter_addr=&(c_thread_state.this_counter().counter_value);
+    // TODO: come up with something better here
+    data.out_uv_counter_addr=reinterpret_cast<uint64_t*>(&(c_thread_state.this_counter().counter_value));
```

Comment:
this is a bit of a hack (but less of a hack that the data race). I imagine the assembly code we generate relies on this exact structure, so this would be quite involved to change.